auth/maintenance_socket_role_manager.cc

```diff
@@ -109,4 +110,12 @@ future<> maintenance_socket_role_manager::remove_attribute(std::string_view role
     return operation_not_supported_exception("REMOVE ATTRIBUTE");
 }
 
+future<std::vector<cql3::description>> maintenance_socket_role_manager::describe_role_grants() {
+    return operation_not_supported_exception<std::vector<cql3::description>>("DESCRIBE SCHEMA WITH INTERNALS");
 }
+
+future<std::vector<cql3::description>> maintenance_socket_role_manager::describe_attached_service_levels() {
```

Comment:
I'm not sure if returning an exception in `maintenance_socket_role_manager::describe_role_grants` and `maintenance_socket_role_manager::describe_attached_service_levels` is a good idea. These functions are unconditionally used in DESCRIBE (all the stuff) variant, so it looks like one won't be able to run it via the maintenance socket. IIRC there were plans for scylla-manager to use the maintenance socket to run DESCRIBE, looks like it will break it.

I think we should have a test for this (whether DESCRIBE works via maintenance socket).

Reply:
Do you suggest we should make `DESCRIBE SCHEMA` work with maintenance socket? At the moment, those functions will never be called because `maintenance_socket_role_manager::query_all()` throws an exception and `DESC SCHEMA` is terminated immediately.

Even if we want to support `DESC SCHEMA` with maintenance socket, I'm not sure if it's a good idea to include it in this PR, especially because the role manager doesn't store any information—we'd simply obtain an empty state of auth. If we wanted to impose other semantics on it, we'd need to design it carefully first, which is another reason why I'm against doing anything with it here.

Reply:
One more problem with the maintenance socket is that AllowAllAuthenticator uses the anonymous user, and it is not a superuser (see `auth::has_superuser`). It will not be possible to dump the salted hashes via the maintenance socket.

Reply:
...and the responsibility to get the password hash of the role is on the authenticator, and AllowAllAuthenticator does not implement it / doesn't make sense to implement this method for it...